radish34/ops/setup_circuits.sh

```diff
@@ -9,18 +9,33 @@ NC='\033[0m'
 docker-compose up --no-recreate -d zkp
 ./ops/await_stack.sh
 
-if [ ! -d ./zkp/output/createMSA ]; then
+if [ "$1" = "" ]; then
```

Comment:
I like this change a lot. However, I think the way it's implemented with circuit selector flags could be better. It feels like the option should be in a config file or somewhere else. Or perhaps, instead of looking for an explicit flag it looks for an environment? Like, `ENV == "CI"` then use dummy?
Or maybe there's a way to swap out the files at the volume level such that the dummy circuits are just aliased or drop-in replacements for the real circuits? Something like you might do with sym links...

I'm not sure I'm explaining this well but there's something about the new `ZKP_MODE` in all the services in the compose files that just seems a bit out of place.

Reply:
To make sure there is a consistent use of a test vs dev mode, one way that could be handled is to store an e.nv file at the root of the repo which can be used across `Dockerfile`/docker-compose environment parameters and `.sh` files.

In terms of symlinks, i believe that could be better handled at the baseline/ level to provide for choosing between test vs real toggle for circuit-set selection. Or should we handle this as part of this pr: create sym-links as additional lines in `ops/build.sh`? Thoughts? @breakpointer @bitwiseguy

Reply:
I spoke with @kthomas yesterday about dynamically creating docker-compose files based on our different environment types (ci vs. dev vs. prod). An example of that is [here](https://github.com/connext/indra/blob/staging/ops/start-dev.sh). I plan on starting a new branch to contain that work. Could we address the circuit selection that way in a separate PR by injecting a "circuit-selection" environment variable to the containers based on the environment type? Or do you all have any thoughts on the dynamically generated docker-compose files in general?

Reply:
agreed, this is a much refined approach to handle custom selections of circuit sets and aligning with the different environments. I concur on addressing circuit-selector work to a separate PR. We can track it as its own issue. Does the approach sound ok to you @breakpointer

Reply:
I like the change also. It makes sense to parameterize which circuit(s) are under test or otherwise required.

Also agree that environment variables should not "leak" out of the service they are intended to be set for.

Reply:
Sure 👍